pkg/ingester/shutdown_marker.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingester
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+)
+
+// ShutdownMarker is responsible for writing a marker file to disk to indicate
+// that an ingester is going to be scaled down in the future. The presence of this
+// file means that an ingester should flush and upload all data when stopping. This
+// file is never cleaned up because the ingester pod will be deleted as part of
+// scaling down.
+type ShutdownMarker struct {
+	Path string
+}
```

Comment:
I don't see why this needs to be a type. Two functions with explicit path parameters are always easier to follow, and more than enough for this case.

This file is also too shallow. There is no need to create one file per type like in Java. Two functions can easily be in ingester.go, especially if we shorten `Create` one.